backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQHealthCheck.java

```diff
@@ -46,27 +49,44 @@ public ComponentName componentName() {
     }
 
     @Override
-    public Result check() {
+    public Mono<Result> checkReactive() {
         try {
-            if (connectionPool.tryConnection() && rabbitChannelPoolImpl.tryChannel()) {
-                Optional<RabbitMQServerVersion> version = connectionPool.version();
+            return Flux.concat(connectionPool.tryConnection(),
+                rabbitChannelPoolImpl.tryChannel())
+                .reduce(true, (a, b) -> a && b)
+                .flatMap(channelOpen -> {
+                    if (channelOpen) {
+                        return checkVersion();
+                    } else {
+                        return Mono.just(Result.unhealthy(COMPONENT_NAME, "The created connection was not opened"));
+                    }
+                })
+                .onErrorResume(e -> Mono.just(Result.unhealthy(COMPONENT_NAME,
+                    "Unhealthy RabbitMQ instances: could not establish a connection", e)));
+        } catch (Exception e) {
+            return Mono.just(Result.unhealthy(COMPONENT_NAME,
+                "Unhealthy RabbitMQ instances: could not establish a connection", e));
+        }
+    }
+
+    private Mono<? extends Result> checkVersion() {
+        return connectionPool.version()
+            .map(Optional::of)
```

Comment:
why do we need some Optional here, can't we handle things with Mono?

Reply:
No as we need the version and if it is acceptable for the error message.